### PR TITLE
Support mongodb-core tracing through the plugin API

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

**Comment by DominicKramer** [comment]
PTAL

File: src/plugins/plugin-mongodb-core.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+var SUPPORTED_VERSIONS = '1 - 2';
+
+function nextWrap(api, next) {
+  return function next_trace(cb) {
+    var root = api.getRootSpan();
+    if (!root) {
+      return next.apply(this, arguments);
+    }
+    var span = api.createChildSpan({
```
Comment (matthewloring) [inline src/plugins/plugin-mongodb-core.js]: Does this object initializer fit on one line?

File: src/plugins/plugin-mongodb-core.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+function wrapWithLabel(api, label) {
+  return function(original) {
+    return function mongo_operation_trace(ns, ops, options, callback) {
+      var root = api.getRootSpan();
+      if (!root) {
+        return original.apply(this, arguments);
+      }
+      var span = api.createChildSpan({
+        name: label
+      });
```
Comment (matthewloring) [inline src/plugins/plugin-mongodb-core.js]: Can we test for a `null` child span here and avoid the root retrieval above?

File: src/plugins/plugin-mongodb-core.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+function nextWrap(api, next) {
+  return function next_trace(cb) {
+    var root = api.getRootSpan();
+    if (!root) {
+      return next.apply(this, arguments);
+    }
+    var span = api.createChildSpan({
+      name: 'mongo-cursor'
+    });
```
Comment (matthewloring) [inline src/plugins/plugin-mongodb-core.js]: Can we test for a null child span here and avoid the root retrieval above?

File: src/plugins/plugin-mongodb-core.js (hunk 1 above)
Comment (DominicKramer) [inline src/plugins/plugin-mongodb-core.js]: Done

File: src/plugins/plugin-mongodb-core.js (hunk 3 above)
Comment (DominicKramer) [inline src/plugins/plugin-mongodb-core.js]: Done

File: src/plugins/plugin-mongodb-core.js (hunk 2 above)
Comment (DominicKramer) [inline src/plugins/plugin-mongodb-core.js]: Done

**Comment by matthewloring** [comment]
LGTM pending CI.

**Pull request merged**